pof/src/write.rs

```diff
@@ -426,256 +433,303 @@ impl Model {
 }
 
 // ==============================================================================
-// DAE Writing
+// DAE / glTF Writing
 // ==============================================================================
 
-use dae_parser::*;
+use dae_parser::{Node as DaeNode, UpAxis::YUp, *};
+
+trait NodeBuilder {
+    type Ctx;
+    type Node;
+    fn children(&mut self) -> &mut Vec<Self::Node>;
+    fn translate(&mut self, val: [f32; 3]);
+    fn rotate(&mut self, axis_angle: ([f32; 3], f32));
+    fn scale(&mut self, val: [f32; 3]);
+    fn matrix_transform(&mut self, mat: Mat4x4);
+    fn build(self, ctx: &mut Self::Ctx) -> Self::Node;
+}
+trait Node {
+    type Ctx;
+    type Builder: NodeBuilder<Node = Self, Ctx = Self::Ctx>;
+    fn new(id: String, name: String) -> Self::Builder;
+    fn new2(id: String) -> Self::Builder {
+        Self::new(id.clone(), id)
+    }
+}
+impl NodeBuilder for DaeNode {
+    type Ctx = ();
+    type Node = Self;
+    fn children(&mut self) -> &mut Vec<DaeNode> {
+        &mut self.children
+    }
+    fn translate(&mut self, val: [f32; 3]) {
+        self.push_transform(Translate::new(val))
+    }
+    fn rotate(&mut self, (axis, angle): ([f32; 3], f32)) {
+        self.push_transform(Rotate::new(axis, angle * (180.0 / PI)))
+    }
+    fn scale(&mut self, val: [f32; 3]) {
+        self.push_transform(Scale::new(val))
+    }
+    fn matrix_transform(&mut self, mat: Mat4x4) {
+        self.push_transform(mat)
+    }
+    fn build(self, _: &mut ()) -> DaeNode {
+        self
+    }
+}
+impl Node for DaeNode {
+    type Ctx = ();
+    type Builder = Self;
+    fn new(id: String, name: String) -> Self {
+        DaeNode::new(id, Some(name))
+    }
+}
 
 // turns a direction vector into a dae Rotate struct
```

Comment:
Obviously, this doesn't do that anymore, so the comment should be updated. Some other comments below referring to dae should also be updated to be more general.